src/library_manager/llext_manager.c

```diff
@@ -272,16 +272,15 @@ uintptr_t llext_manager_allocate_module(struct processing_module *proc,
 	return mod_manifest->module.entry_point;
 }
 
-int llext_manager_free_module(struct processing_module *proc,
-			      struct comp_ipc_config *ipc_config)
+int llext_manager_free_module(const uint32_t component_id)
 {
 	struct sof_man_fw_desc *desc;
 	struct sof_man_module *mod;
-	uint32_t module_id = IPC4_MOD_ID(ipc_config->id);
+	const uint32_t module_id = IPC4_MOD_ID(component_id);
```

Comment:
see, in the end all you need is the module ID, so you can also use that here as the argument

Reply:
this contains also a lib id.